### PR TITLE
Issue 6

### LINKED ISSUE #6: BE: Auth
- simple password validation (>=6 chars)
- Use trusted library for hashing and verifying (bcrypt)
- Implement a middleware for protected routes ( checks token, returns appropriate response if invalid)
- Use httpOnly cookies 
- On successful auth: return token in the cookies and user object

### PR DESCRIPTION
**Pull request by InsafKhamzin**
Closes #6 
/auth/signup and /auth/signin routes implemented
New .env vars (values for development environment):
- JWT_SECRET=secret
- JWT_EXPIRATION=30d
- COOKIE_EXPIRATION=30

Logic is divided into separate folders:
/models - db mongoose models of the app
/services - the main login of app
/validators - requests from frontend are validated here
/utils - some helpers for different purposes

### CODE REVIEW

**Comment by darrenMabbayad** [comment]
I merged this into my front-end branch and I've been testing the routes. However, both in the react app and in Postman, I'm getting error code 500 on both auth routes. I am able to add new users to the database, and the routes are getting the request body params, but I'm not sure what could be the problem.

File: server/routes/auth.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// @param password
+
+router.post("/signup", signup, async function (req, res, next) {
+  try {
+    const { newUser, token, cookieConfig } = await authService.signUpUser(
+      req.body
+    );
+    return res.status(201).cookie("token", token, cookieConfig).json(newUser);
+  } catch (error) {
+    next(error);
```
Comment (aeciorc) [inline server/routes/auth.js]: Any specific reason to catch this error here vs let be picked up by the error handler?

**Review by aeciorc — APPROVED** [review]
Solid PR 👍 
One suggestion: make the name of the branches and PR more descriptive, e.g "backend-auth", "Add authentication routes and auth middleware"

**Review by darrenMabbayad — APPROVED** [review]
all the routes and error handling looks good to me, and I can access all the response data from the frontend

**Comment by InsafKhamzin** [comment]
Added express-async-errors module which allows using async middlewares. It patches express and you don't have to use next() or asyncHandler in every controller

**Review by Hyunse — APPROVED** [review]
I deleted some comments because there were validation check in validator! loosk good

**Pull request merged**